### PR TITLE
PUB-2564 Move Data-management functional tests into integration test folder

### PR DESCRIPTION
**Pull request by KianKwa**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-2564

### Change description ###

- Move current data-management functional tests to integration test folder.
- Move functional test prototype test code into this branch.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/UploadPublicationTest.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.github.dockerjava.zerodep.shaded.org.apache.hc.core5.http.HttpHeaders.AUTHORIZATION;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.http.HttpStatus.CREATED;
+import static uk.gov.hmcts.reform.pip.data.management.utils.TestUtil.BEARER;
+import static uk.gov.hmcts.reform.pip.data.management.utils.TestUtil.randomLocationId;
+
+
```
Comment (ChrisS1512) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/UploadPublicationTest.java:34]: Extra space here :)

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/AuthException.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.pip.data.management.utils;
+
+@SuppressWarnings("PMD.JUnit5TestShouldBePackagePrivate")
```
Comment (ChrisS1512) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/AuthException.java:3]: Can I double check the Suppression here is needed?

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/FunctionalTestBase.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        return given()
+            .relaxedHTTPSValidation()
+            .headers(getRequestHeaders(additionalHeaders))
+            .body(body)
+            .when()
+            .delete(path)
+            .thenReturn();
+    }
+
+    private static Map<String, String> getRequestHeaders(final Map<String, String> additionalHeaders) {
```
Comment (ChrisS1512) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/FunctionalTestBase.java:72]: I don't believe this method is needed / not sure we will need it in the future.

Do you agree and if so can this be removed?

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/OAuthClient.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            .baseUri("https://login.microsoftonline.com/" + tenantId + "/oauth2/v2.0/token")
+            .post()
+            .body()
+            .jsonPath()
+            .get("access_token");
+
+        if (token == null) {
+            throw new AuthException(
+                String.format(
+                    "Unable to get token with %s %s %s %s %s", clientId, clientSecret, tenantId, scope
```
Comment (ChrisS1512) [inline src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/OAuthClient.java:40]: We should not log the client details here. I think something like Unable to get token for the API or similar should work.

File: src/integrationTest/resources/application-integration.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+spring:
+  cloud:
+    azure:
```
Comment (ChrisS1512) [inline src/integrationTest/resources/application-integration.yaml:3]: I wonder if we need this azure part as seems similar to the main app yaml. If not already could you try removing and see if this still works?

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]
There's a Mock blob file that exists within the functionalTest folder that I don't believe is needed anymore - AzureBlobConfigurationTestConfiguration

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/AuthException.java (hunk 2 above)
Comment (KianKwa) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/AuthException.java]: Removed suppression

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/UploadPublicationTest.java (hunk 1 above)
Comment (KianKwa) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/UploadPublicationTest.java]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/FunctionalTestBase.java (hunk 3 above)
Comment (KianKwa) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/FunctionalTestBase.java:71]: This is used by other public method within the class

File: src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/OAuthClient.java (hunk 4 above)
Comment (KianKwa) [reply src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/utils/OAuthClient.java]: Updated exception message

File: src/integrationTest/resources/application-integration.yaml (hunk 5 above)
Comment (KianKwa) [reply src/integrationTest/resources/application-integration.yaml]: Removed

**Pull request closed without merging**